dask_expr/_collection.py

```diff
@@ -3098,17 +3105,9 @@ def __array__(self):
 
 def new_collection(expr):
     """Create new collection from an expr"""
-
     meta = expr._meta
     expr._name  # Ensure backend is imported
-    if is_dataframe_like(meta):
-        return DataFrame(expr)
-    elif is_series_like(meta):
-        return Series(expr)
-    elif is_index_like(meta):
-        return Index(expr)
-    else:
-        return Scalar(expr)
+    return get_collection_type(meta)(expr)
```

Comment:
Note that we were have been using `new_collection` from the beginning, because we know we needed to make this change to support other backends (i.e. cudf) long term.

The behavior of `new_collection` is now consistent with the behavior of `new_dd_object` in `dask.dataframe`.